include/fluent-bit/flb_event_loop.h

```diff
@@ -76,27 +76,26 @@ static inline void flb_event_load_injected_events(struct flb_bucket_queue *bktq,
     int __flb_event_priority_live_foreach_n_events = evl->n_events;                     \
     for (                                                                               \
         /* init */                                                                      \
-        flb_event_load_bucket_queue(bktq, evl),                                         \
-        event = flb_bucket_queue_find_min(bktq) ?                                       \
-                mk_list_entry(                                                          \
-                    flb_bucket_queue_pop_min(bktq), struct mk_event, _priority_head) :  \
-                NULL;                                                                   \
+        flb_event_load_bucket_queue(bktq, evl);                                         \
                                                                                         \
         /* condition */                                                                 \
-        event != NULL &&                                                                \
-        (__flb_event_priority_live_foreach_iter < max_iter || max_iter == -1);          \
+        (__flb_event_priority_live_foreach_iter < max_iter || max_iter == -1)           \
+        && (NULL != (                                                                   \
+            event = flb_bucket_queue_find_min(bktq)                                     \
+                    ? mk_list_entry(flb_bucket_queue_pop_min(bktq),                     \
+                                    struct mk_event,                                    \
+                                    _priority_head)                                     \
+                    : NULL                                                              \
+        ));                                                                             \
```

Comment:
We can refactor this later, but it think this section would be more readable if we had it as:
```
        (__flb_event_priority_live_foreach_iter < max_iter || max_iter == -1)           \
        && !flb_bucket_queue_is_empty(bktq)                                             \
        && (event = mk_list_entry(flb_bucket_queue_pop_min(bktq),                       \
                                    struct mk_event,                                    \
                                    _priority_head))

Reply:
I don't understand this, why would we want to get the event in the check condition for the for loop??

Also, the new condition is added with logical AND `&&`, then won't this mean the loop will still terminate when the first `__flb_event_priority_live_foreach_iter < max_iter` returns false, and the statement after the AND won't run. By which I mean- I don't understand the difference between this and making the new bits be the first statements in the loop. (As in not part of the for condition definition). 

??

Reply:
Here's how the for loop parts are run:
1. Init
2. Condition
3. Code
4. Update
5. Condition

If you take an event out in the update and the condition aborts, then you risk loosing one event
1. Update       - take event out of bucket queue 
2. Condition   - potentially abort, which gets rid of the event.


With this code change:
We don't remove the next event from the bucket queue if max-iter is reached.

Reply:
Ok I think I might get it now...

Reply:
https://stackoverflow.com/questions/33457399/when-does-the-loop-variable-in-a-for-loop-get-updated